slides.Rmd

```diff
@@ -419,54 +438,76 @@ Borders %>%
 ```
 
-```{r,  eval=FALSE,echo=TRUE}
-recode(Borders$location, "B120V" = "B120H")
+## rename() 
+
+The dplyr package also makes it very straight forward to rename specific columns by using the rename() function respectively. The rename() function works as follows: 
+
+```{r, eval=FALSE,echo=TRUE}
+df <- rename(df, new_code = existing_code)
 ```
 
+```{r, eval=FALSE,echo=TRUE}
+Borders <- rename(Borders, DateOfBirth = Dateofbirth)
+```
 
-## Exercise
+If you look at your Borders dataframe, you will see that the Dateofbirth column is now called DateOfBirth. 
 
-- Select the URI, Specialty. and Dateofbirth columns from Borders data. Arrange the data in ascending order by Specialty. Use the filter function to count how many records have a missing Dateofbirth value.
+## recode() 
 
-## Merging Data 
+While rename() changes the column name, recode() changes the values within the column. The recode() function works best when used within a mutate(): 
 
-The dplyr package also makes it very easy to merge data by matching files together using common variables. There are several join functions within dplyr that are designed to merge dataframes together to create one dataframe containing the relevant variables.
-The general structure of a join function is to specify the two dataframes used for merging, x and y, and a common variable to match by. For example, for an inner_join you would have: 
+```{r,  eval=FALSE,echo=TRUE}
+df <- df %>% 
+  mutate(column_name = recode(column_name, 
+                              "existing_code" = "new_code"))
+```
 
 ```{r,  eval=FALSE,echo=TRUE}
-inner_join(x = dataframe1, y = dataframe2, by = "common_variable")
+Borders <- Borders %>% 
+  mutate(HospitalCode = recode(HospitalCode, "B120V" = "B120H"))
 ```
 
-## Examples of Merges
 
-.	left_join(x, y) - join x and y, by keeping all rows in x and all columns in both x and y, and merging based on common variables
+## Exercise 3
 
-.	right_join(x, y) - join x and y, by keeping all rows in y and all columns in both x and y, and merging based on common variables
+- Select the URI, Specialty. and DateOfBirth columns from Borders data. Arrange the data in ascending order by Specialty. Use the filter function to count how many records have a missing Dateofbirth value.
 
-.	inner_join(x, y) - join x and y, by keeping all rows that exist in both x and y and all columns from both x and y
+## Joining Data 
 
-.	full_join(x, y) - join x and y, by keeping all rows and columns from both x and y, and merging based on common variables
+The dplyr package also makes it very easy to merge data by matching files together using common variables. There are several join functions within dplyr that are designed to merge dataframes together to create one dataframe containing the relevant variables.
+The general structure of a join function is to specify the two dataframes used for merging, x and y, and a common variable to match by. For example, for an inner_join you would have: 
```

Comment:
Last line should be left_join not inner_join

Reply:
Thanks James, I have pushed a corrected version of the markdown file.